File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        StringBuilder r = new StringBuilder();
        String s = strs[0];
        for(String x : strs){
            if(s.length() > x.length()) s = x;
        }
        for(int i = 0 ; i < s.length() ; i++){
            char c = s.charAt(i);
            for(int j = 0 ; j < strs.length; j++){
                if(c != strs[j].charAt(i)) return r.toString();
            }
            r.append(c);
        }
        return r.toString();
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        ans = ""
        flag = True
        idx = 0
        while flag:
            c = strs[0][idx] if idx < len(strs[0]) else None
            for s in strs:
                if not (idx < len(s) and s[idx] == c):
                    flag = False
            if flag:
                idx += 1
                ans += c
        return ans
                
                
        